vpr/src/route/rr_graph_fpga_interchange.cpp

```diff
@@ -0,0 +1,1179 @@
+#include <kj/std/iostream.h>
+#include <zlib.h>
+#include <algorithm>
+#include <map>
+#include <queue>
+
+#include "rr_graph_fpga_interchange.h"
+
+#include "arch_util.h"
+#include "vtr_time.h"
+#include "vtr_error.h"
+#include "globals.h"
+#include "check_rr_graph.h"
+#include "rr_metadata.h"
+#include "rr_graph_indexed_data.h"
+#include "fpga_interchange_arch_utils.h"
+
+using namespace DeviceResources;
+using namespace LogicalNetlist;
+using namespace capnp;
+
+typedef std::tuple<int, int> location;
+
+/*
+ * Intermediate data type.
+ * It is used to build and explore FPGA Interchange
+ * nodes graph and later convert them to rr_nodes.
+ */
+struct node_ {
```

Comment:
This may need to be renamed to something else, just to avoid confusion on which kind of node it represents. Maybe something like `node_data` or `intermediate_node`